lms/djangoapps/experiments/views.py

```diff
@@ -47,3 +54,31 @@ def create_or_update(self, request, *args, **kwargs):
 
         self.action = 'create'
         return self.create(request, *args, **kwargs)
+
+    def _cache_users(self, usernames):
+        users = User.objects.filter(username__in=usernames)
+        self._cached_users = {user.username: user for user in users}
+
+    def _get_user(self, username):
+        user = self._cached_users.get(username)
+
+        if not user:
+            user = User.objects.get(username=username)
+            self._cached_users[username] = user
+
+        return user
+
+    @list_route(methods=['put'], permission_classes=[permissions.IsAdminUser])
+    def bulk_upsert(self, request):
+        upserted = []
+        self._cache_users([datum['user'] for datum in request.data])
+
+        with transaction.atomic():
+            for item in request.data:
+                user = self._get_user(username=item['user'])
+                datum, __ = ExperimentData.objects.update_or_create(
```

Comment:
The fields in use are indexed, and the table size is quite small. I don't think we will have any performance issues. If we do need to batch, I'd rather do so in the code that makes the API calls than spend more time on the API itself (in the spirit of moving rapidly).